ZenPacks/zenoss/OpenStackInfrastructure/events.py

```diff
@@ -922,34 +922,39 @@ def event_is_mapped(evt):
     return evt.eventClassKey in MAPPERS
 
 
+def map_event(evt):
+    # given an event, return an ObjectMap, or None.
+
+    (idfunc, mapper) = MAPPERS.get(evt.eventClassKey, (None, None))
+
+    if idfunc:
+        component_id = idfunc(evt)
+        if component_id is None:
+            LOG.error("Unable to identify component ID for %s event: %s",
+                      evt.eventClassKey, evt.getStateToCopy())
+            return None
+        evt.component = idfunc(evt)
+
+    if mapper:
+        return mapper(evt)
+
+
 # ==============================================================================
 #  This is the main process() function that is called by the Events Transform
 #  subsystem. See objects.xml for further detail.
 # ==============================================================================
-
 def process(evt, device, dmd, txnCommit):    
     try:
-        (idfunc, mapper) = MAPPERS.get(evt.eventClassKey, (None, None))
-
-        if idfunc:
-            component_id = idfunc(evt)
-            if component_id is None:
-                LOG.error("Unable to identify component ID for %s event: %s",
-                          evt.eventClassKey, evt.getStateToCopy())
-                return 0
-            evt.component = idfunc(evt)
-
-        if mapper:
-            datamaps = mapper(evt)
-            if datamaps:
-                adm = ApplyDataMap(device)
-                for datamap in datamaps:
-                    # LOG.debug("Applying %s" % datamap)
-                    adm._applyDataMap(device, datamap)
-
-            return len(datamaps)
+        objectmap = map_event(evt)
+
+        if objectmap:
+            adm = ApplyDataMap(device)
+            # LOG.debug("Applying %s" % objectmap)
```

Comment:
We should either expose the LOG or remove the comment.

Reply:
Yeah, that's old code that I didn't change, but i'll fix it.